airbyte-cdk/python/airbyte_cdk/entrypoint.py

```diff
@@ -232,10 +235,24 @@ def _emit_queued_messages(self, source: Source) -> Iterable[AirbyteMessage]:
 def launch(source: Source, args: List[str]) -> None:
     source_entrypoint = AirbyteEntrypoint(source)
     parsed_args = source_entrypoint.parse_args(args)
-    for message in source_entrypoint.run(parsed_args):
-        # simply printing is creating issues for concurrent CDK as Python uses different two instructions to print: one for the message and
-        # the other for the break line. Adding `\n` to the message ensure that both are printed at the same time
-        print(f"{message}\n", end="", flush=True)
+    messages = iter(source_entrypoint.run(parsed_args))
+
+    if parsed_args.flush == 0:
+        for message in messages:
+            print(f"{message}\n", end="", flush=True)
+        return
+
+    while True:
+        buffer = io.StringIO()
+        end_time = time.monotonic() + parsed_args.flush
+        try:
+            while time.monotonic() < end_time:
+                message = next(messages)
+                buffer.write(f"{message}\n")
+        except StopIteration:
+            break
+        finally:
+            print(buffer.getvalue(), end="", flush=True)
```

Comment:
Generally looks good and I can confirm that it does speed up!

Could you please refactor to:

1. catch ALL output, not only `next(messages)`: `logger` is used throughout the code, and we need to keep the same order of messages.
2. use `context manager` to encapsulate buffering logic.

Reply:
1. Done but it might make the suggested improvement a bit more dangerous than anticipated.
2. Good point, done